src/Ditto.Kinesis/ReplicatingConsumer.cs

```diff
@@ -0,0 +1,128 @@
+using Prometheus;
+using Serilog.Events;
+using SerilogTimings.Extensions;
+using ILogger = Serilog.ILogger;
+
+namespace Ditto.Kinesis
+{
+    /// <summary>
+    /// Stream consumer that replicates to the destination event store
+    /// </summary>
+    public class ReplicatingConsumer : ICompetingConsumer
+    {
+        private readonly IAmazonKinesis _kinesis;
+        private readonly KinesisSettings _kinesisSettings;
+        private readonly Serilog.ILogger _logger;
+        private readonly DittoSettings _dittoSettings;
+
+        public ReplicatingConsumer(IAmazonKinesis kinesis, KinesisSettings kinesisSettings, DittoSettings dittoSettings, ILogger logger, string streamName, string groupName)
+        {
+            _kinesis = kinesis ?? throw new ArgumentNullException(nameof(kinesis));
+            _kinesisSettings = kinesisSettings ?? throw new ArgumentNullException(nameof(kinesisSettings));
+            _dittoSettings = dittoSettings ?? throw new ArgumentNullException(nameof(dittoSettings));
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+
+            StreamName = streamName ?? throw new ArgumentNullException(nameof(streamName));
+            GroupName = groupName ?? throw new ArgumentNullException(nameof(groupName));
+        }
+
+        public string StreamName { get; }
+        public string GroupName { get; }
+        public bool CanConsume(string eventType) => true;
+
+        public async Task ConsumeAsync(string eventType, ResolvedEvent resolvedEvent)
+        {
+            if (string.IsNullOrWhiteSpace(eventType)) throw new ArgumentException("Event type required", nameof(eventType));
+
+            if (_dittoSettings.ReadOnly)
+            {
+                _logger.Debug("Received {EventType} #{EventNumber} from {StreamName} (Original Event: #{OriginalEventNumber})",
+                    resolvedEvent.Event.EventType,
+                    resolvedEvent.Event.EventNumber,
+                    resolvedEvent.Event.EventStreamId,
+                    resolvedEvent.OriginalEventNumber
+                );
+
+                return;
+            }
+
+            var request = new PutRecordRequest
+            {
+                StreamName = _kinesisSettings.StreamName,
+                PartitionKey = resolvedEvent.Event.EventStreamId,
+                Data = CreateDataStream(resolvedEvent)
+            };
+
+            using (_logger.OperationAt(LogEventLevel.Debug).Time("Replicating {EventType} #{EventNumber} from {StreamName} (Original Event: #{OriginalEventNumber}) to Kinesis",
+                resolvedEvent.Event.EventType,
+                resolvedEvent.Event.EventNumber,
+                resolvedEvent.Event.EventStreamId,
+                resolvedEvent.OriginalEventNumber))
+            using (DittoMetrics.IODuration.WithIOLabels("kinesis", _kinesisSettings.StreamName, "put_record").NewTimer())
+            {
+                await _kinesis.PutRecordAsync(request);
```

Comment:
We ideally want to batch write multiple events.

Reply:
Is there any retry strategies here? You'll see two exceptions in normal operation, `ProvisionedThroughputExceededException` and occasionally `AmazonKinesisException` when something goes wrong in AWS systems. Typically I catch all exceptions around put and just do a simple Polly exponential backoff with jitter, could be useful in the case of having many Dittos.

Reply:
As discussed, we can use the Standard retry mode by setting an environment variable in AWS